### ISSUE TITLE
Cache is not URL-aware

### ISSUE THREAD

**Issue by flyte** [inirudebwoy/interview-task#3]
What happens if we change URLs to maybe a different service with different information (but conveniently the same website layout..)? Then we'd have to tie the cached information to a specific URL so that we don't return cached results across different datasets.

**Comment by inirudebwoy** [comment]
True. According to README we should only get results from http://www.socialsecurity.gov/OACT/babynames/ but we both made code our code to accept any URL. We either hardcode URL or accept any. I don't think it is possible to create scraper that would parse any website for data.

Thus we can remove URL parameter from code and use global or curry every function that accepts url as argument with prepared URL.
